internal/controller/kustomization_controller.go

```diff
@@ -812,7 +813,26 @@ func (r *KustomizationReconciler) apply(ctx context.Context,
 		return false, nil, err
 	}
 
-	if cmeta := obj.Spec.CommonMetadata; cmeta != nil {
+	var revisionAnnotations map[string]string
+	if r.EnableRevisionAnnotation {
+		if obj.Status.LastAppliedRevision == "" || revision != obj.Status.LastAppliedRevision {
+			revisionAnnotations = map[string]string{
+				fmt.Sprintf("%s/source-revision", eventv1.Group): revision,
+			}
+		}
+		if originRevision != "" && originRevision != obj.Status.LastAppliedOriginRevision {
+			revisionAnnotations[fmt.Sprintf("%s/source-revision", eventv1.Group)] = originRevision
+		}
+	}
+
+	cmeta := obj.Spec.CommonMetadata
+	if cmeta != nil {
+		if cmeta.Annotations == nil {
+			cmeta.Annotations = make(map[string]string)
+		}
+		for k, v := range revisionAnnotations {
+			cmeta.Annotations[k] = v
```

Comment:
Why are we annotating all resources? A Kubernetes deployment will never send an event to NC. This will cause drift for potentially thousands of non-Flux resources, which negates one of the core Flux design principles, we never alter the cluster state, unless a change is made in Git; in this case, no change was made in Git, you just blindly mutate all objects.